Code/Framework/AzFramework/Tests/Platform/Common/Xcb/XcbInputDeviceKeyboardTests.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <xcb/xcb.h>
+
+#include <AzFramework/XcbApplication.h>
+#include <AzFramework/XcbInputDeviceKeyboard.h>
+#include "MockXcbInterface.h"
+#include "Actions.h"
+
+template<typename T>
+xcb_generic_event_t MakeEvent(T event)
+{
+    return *reinterpret_cast<xcb_generic_event_t*>(&event);
+}
+
+namespace AzFramework
+{
+    TEST(XcbInputDeviceKeyboard, InputChannelsUpdateStateFromXcbEvents)
```

Comment:
We should always use a Test Fixture for our UnitTest.

Most likely in the future this test will need AZ Allocators and it is simple enough to alias UnitTest::ScopedAllocatorSetupFixture.

```suggestion
    using XcbInputDeviceKeyboard = UnitTest::ScopedAllocatorFixture
    TEST_F(XcbInputDeviceKeyboard, InputChannelsUpdateStateFromXcbEvents)
```

I think it is also better to place the test in the UnitTest namespace, but that is only a preference.

Reply:
This test uses `AzFramework::Application`, which is responsible for setting up the allocators. I can see some of the expectations of the XcbInterface mock being made in a fixture, so that other tests can reuse similar expectations, but I don't know what those tests are yet. I think it is simple enough to define that fixture as we need it in the future.